src/features/Admin/Rooms/AddNewRoom/AddNewRoom.tsx

```diff
@@ -1,9 +1,407 @@
-import React from 'react'
+import { Button, Checkbox, Divider, FormControl, Grid, ListItemText, Paper, Select, TextField } from "@mui/material";
+import { Box, Container } from "@mui/system";
+import { useContext, useState, useEffect } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import Styles from "./AddNewRoom.module.scss";
+import { InputLabel, MenuItem } from "@mui/material";
+import axios from "axios";
+import { AuthContext } from "../../../../context/AuthContext";
+import { addroomsUrl } from "../../../../services/api";
+import { IAddRoom } from "../../../../interface/RoomInterface";
+import { faciRoomsUrl } from "./../../../../services/api";
+import DragDropFileUpload from "../../../../shared/DragDropFileUpload/DragDropFileUpload";
+const AddNewRoom: React.FC = () => {
+  const { requestHeaders } = useContext(AuthContext);
+
+  const navigate = useNavigate();
+
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [facilitiesList, setFacilitiesList] = useState([]);
+  const [selectedImage, setSelectedImage] = useState(null);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    formState: { errors },
+
+  } = useForm<IAddRoom>();
+
+  // Format Data
+  const appendToFormData = (data: IAddRoom) => {
+    const formData = new FormData();
+    formData.append("roomNumber", data["roomNumber"]);
```

Comment:
You can access properties of objects using the name directly without ["propertyName"]:
For example `data?.roomNumber` instead of `data["roomNumber"]`